pom.xml

```diff
@@ -83,11 +88,13 @@
                         <goals><goal>run</goal></goals>
                         <configuration>
                             <target>
+                                <echo message="download archive" />
+                                <get src="${upstream.url}" dest="${project.build.directory}/${project.artifactId}.zip" />
```

Comment:
It is fine to do the download this way but you should remove the `wagon-maven-plugin`.